jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/ApplyService.java

```diff
@@ -112,12 +118,15 @@ public class ApplyService {
     private boolean rollingUpgradePreserveScale = true;
     private boolean recreateMode;
     private final PatchService patchService;
+    private final IngressControllerDetectorService ingressControllerDetectorService;
     // This map is to track projects created.
     private static final Set<String> projectsCreated = new HashSet<>();
 
     public ApplyService(KubernetesClient kubernetesClient, KitLogger log) {
         this.kubernetesClient = kubernetesClient;
         this.patchService = new PatchService(kubernetesClient, log);
+        this.ingressControllerDetectorService = new IngressControllerDetectorService(log);
+        initializeControllerDetectors(ingressControllerDetectorService);
```

Comment:
We want to inject the dependencies here.
This should also allow for better testing.

Reply:
I'm not sure I get it. Could you please elaborate?

Reply:
your instantiating the variable. What we want is to pass them into the class